### ISSUE TITLE
Flaky test in crypto package: TestDSA

### ISSUE THREAD

**Issue by hkparker** [hkparker/go-i2p#12]
Every so often the `TestDSA` test fails

```
?	github.com/hkparker/go-i2p	[no test files]
ok      github.com/hkparker/go-i2p/lib/common   0.007s  coverage: 77.8% of statements
?       github.com/hkparker/go-i2p/lib/common/base32    [no test files]
?       github.com/hkparker/go-i2p/lib/common/base64    [no test files]
?       github.com/hkparker/go-i2p/lib/config   [no test files]
--- FAIL: TestDSA (0.00s)
        dsa_test.go:32: sig="R\x06D\x97\x0fu#\xd1Ҧ\xf3\x11G\x0e\x0e^\xde\ue445\x91\xaf+7\u007f\xbf\xaeC\x1eͩ\x9ctsS\x90\x9b\x0fY\x00"
        dsa_test.go:43: failed: invalid signature
FAIL
coverage: 55.0% of statements
FAIL    github.com/hkparker/go-i2p/lib/crypto   0.065s
?       github.com/hkparker/go-i2p/lib/i2np     [no test files]
?       github.com/hkparker/go-i2p/lib/netdb    [no test files]
?       github.com/hkparker/go-i2p/lib/router   [no test files]
?       github.com/hkparker/go-i2p/lib/transport        [no test files]
?       github.com/hkparker/go-i2p/lib/transport/ssu    [no test files]
?       github.com/hkparker/go-i2p/lib/tunnel   [no test files]
```

**Comment by hkparker** [comment]
@majestrate Have you ever seen this happen?

**Comment by majestrate** [comment]
I forget if the DSA code checks for q < p (or whatever that constraint for dsa is), that is possibly it.

**Comment by majestrate** [comment]
found it, the problem was that `big.Int` is encoded as bytes differently than i thought, sometimes r or s is < 20 bytes long so they'd have to be encoded differently.

**Comment by majestrate** [comment]
I think i see a similar problem in elg too

**Comment by hkparker** [comment]
merged #13, thanks for taking a look
